cirq/sim/density_matrix_simulator.py

```diff
@@ -13,202 +13,121 @@
 # limitations under the License.
 """Simulator for density matrices that simulates noisy quantum circuits."""
 
-import collections
-
-from typing import Any, Dict, Iterator, List, TYPE_CHECKING, Tuple, Type, Union
+from typing import Any, Dict, List, TYPE_CHECKING, Tuple, Type, Union
 
 import numpy as np
 
-from cirq import circuits, ops, protocols, qis, study, value, devices
+from cirq import ops, protocols, qis, study, value
 from cirq.sim import density_matrix_utils, simulator
-from cirq.sim.simulator import check_all_resolved
+from cirq.sim.abstract_state import AbstractState
+from cirq.sim.op_by_op_simulator import StateFactory, SimulationResultFactory, OpByOpSimulator
```

Comment:
Prefer importing the full module, rather than individual objects / methods: [style guide](https://github.com/quantumlib/Cirq/blob/master/docs/dev/style.md).

Similarly for other files in this PR.